prusa-slicer: 2.1.1 -> 2.2.0

Moredread opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change

Update to latest version, `2.2.0`. Several new dependencies and a bit of cleanup.

cc @mweinelt @ArdaXi @baracoder @thorstenweber83 as you were interested in the previous PRs

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Moredread commented:
@GrahamcOfBorg build prusa-slicer

Moredread commented:
@GrahamcOfBorg build prusa-slicer

mweinelt commented:
Tested through `nixpkgs-review pr` and saw no compilation issues. Did a quick run test and things looked fine, but I don't have access to our 3D printer currently, since we shutdown our hackerspace.

Moredread commented:
@GrahamcOfBorg build prusa-slicer

Moredread commented:
@GrahamcOfBorg build prusa-slicer

Moredread commented:
@GrahamcOfBorg build prusa-slicer

Moredread commented:
@mweinelt can you have a look again please, now that the final release is up?

mweinelt commented:
Building looks good, when starting I immediately get

```
[2020-03-23 14:36:05.454724] [0x00007f88637fe700] [error]   Unable to initialize hidapi library: failed to load libudev.so.1 or libudev.so.0
```

Does that affect the USB connection to the printer? I don't know yet.

Everything else looks fine to me.

Moredread commented:
@mweinelt I'm not sure, that is new I think. I also don't have a printer atm.

Moredread commented:
I think it works now. :)

Mic92 commented:
@mweinelt can you test again?

mweinelt commented:
Yep, error is gone. Everything else lgtm.

Mic92 commented on pkgs/applications/misc/prusa-slicer/default.nix:
```diff
@@ -74,6 +76,11 @@ stdenv.mkDerivation rec {
 
+  preFixup = stdenv.lib.optionalString stdenv.isLinux ''
+    wrapProgram $out/bin/prusa-slicer \
+      --prefix LD_LIBRARY_PATH : "${systemd.lib}/lib"
```
If you add systemd to the `buildInputs` and also set

```
NIX_LDFLAGS = "-ludev";
```

then you can force link `libudev` into `prusa-slicer` and make it also available to `dlopen`.
This generally causes less problems than `LD_LIBRARY_PATH` because the latter one is also inherited by child processes that might be linked against incompatible libraries.

Moredread replied:
Changed it :)

Moredread committed:
prusa-slicer: 2.1.1 -> 2.2.0

Adding myself (moredread) as maintainer. Several new dependencies and a bit of cleanup.
